cpp/daal/src/data_management/finiteness_checker_avx2_impl.i

```diff
@@ -0,0 +1,234 @@
+/** file: finiteness_checker_avx2_impl.i */
+/*******************************************************************************
+* Copyright contributors to the oneDAL project
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#ifndef __FINITENESS_CHECKER_AVX2_IMPL_I__
+#define __FINITENESS_CHECKER_AVX2_IMPL_I__
+
+template <>
+float sumWithAVX<float, avx2>(size_t n, const float * dataPtr)
```

Comment:
I think further there might be non-AVX implementation of those functions. We already have 3 SIMD variants supported: SSE, AVX and Arm's SVE. There might be more in the future.

```suggestion
float sumWithSIMD<float, avx2>(size_t n, const float * dataPtr)
```